SampleGoGrpc/SimpleMath/grpc.go

```diff
@@ -0,0 +1,259 @@
+//Generated by GRPC Go plugin
+package SimpleMath
+
+import "github.com/google/flatbuffers/go"
+
+import (
+  grpc "google.golang.org/grpc"
+  context "golang.org/x/net/context"
+)
+
+// FlatCodec implements Grpc-go Codec
+
+type FlatCodec struct{}
+
+func (FlatCodec) Marshal(v interface{}) ([]byte, error) {
+  return v.(*flatbuffers.Builder).FinishedBytes(), nil
+}
+
+func (FlatCodec) Unmarshal(data []byte, v interface{}) error {
+  v.(flatbuffersInit).Init(data, flatbuffers.GetUOffsetT(data))
+  return nil
+}
+
+func (FlatCodec) String() string {
+  return "flatbuffers"}
+
+type flatbuffersInit interface {
+  Init(data []byte, i flatbuffers.UOffsetT)
+}
+
+// Client API for Math service
+type MathClient interface{
+  Mod2(ctx context.Context, in *flatbuffers.Builder, opts... grpc.CallOption) (* NumRes, error)  
+  AddNumbers(ctx context.Context, opts... grpc.CallOption) (Math_AddNumbersClient, error)  
+  Squares(ctx context.Context, in *flatbuffers.Builder, opts... grpc.CallOption) (Math_SquaresClient, error)  
+  IsOdd(ctx context.Context, opts... grpc.CallOption) (Math_IsOddClient, error)  
+}
+
+type mathClient struct{
```

Comment:
mathClient is returned as an interface by the method NewMathClient, more over it is private to avoid unnecessary public names as it would never be accessed directly. Same convention is followed by the protobuf grpc-go plugin